ocean_lib/ocean/ocean_assets.py

```diff
@@ -374,10 +459,9 @@ def create(
             from_wallet=publisher_wallet,
         )
 
-        # Fetch the asset on chain
-        asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
+        self._web3.eth.wait_for_transaction_receipt(tx_result)
 
-        return asset
+        return wait_for_asset_update(self._get_aquarius(), asset, tx_result)
```

Comment:
I think you can assign an aquarius instance to avoid repetition of the get aquarius part here and above.